digits/static/js/table_selection.js

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2015, NVIDIA CORPORATION.  All rights reserved.
+
+// class to setup the event handling to add extended selection for a
+// dataTable. This allows multiple dataTables to function on the same
+// page. This adds mouse/pointer based click and marquee selection
+// with shift and ctrl/meta modification for range and toggle
+// selection, up/down arrow key also with shift and ctrl/meta
+// modification, and alt-a, alt-d, and alt-i for select-all,
+// deselect-all, and invert-selection.
+
+// Because key events seem to need to go through the document, the
+// current table is tracked so that the correct table can be
+// affected. There's a chance that would fail if someone could put a
+// subdataTable inside of another dataTable, but that's unlikely if
+// not impossible.
+
+// var table = $('#my_table').DataTable();
+// var ts = new TableSelection('#my_table');
+
+function TableSelection(table_id) {
+    // class state variables
+    this.last_selected_row = null;
+    this.mousedown_row = null;
+    this.mousedown_with_alt = false;
+    this.mousedown_with_shift = false;
+    this.mousedown_was_selected = false;
+    this.mousedown_last_rowIndex = -1;
+
+    function on_selection_changed() {
+        var e = $.Event('selection_changed');
+        e.count = $(TableSelection.current_table + ' tr.selected').length;
+        $(window).trigger(e);
+    }
+
+    // Disablew text selection because we want to select rows not text
+    $(table_id + ' tbody').disableSelection();
```

Comment:
You are correct. You can not copy and paste. That is the trade off, but I see this style of selection far more useful.  If you notice, I took over Alt-A (Ctrl-A) select all text when the pointer is within the rectangle of a table.